lib/grpc/stub.ex

```diff
@@ -150,11 +150,18 @@ defmodule GRPC.Stub do
     parse_unary_response(response, unmarshal)
   end
 
-  defp parse_unary_response({_headers, data_list}, unmarshal) do
-    data_list
-    |> Enum.join("")
-    |> GRPC.Message.from_data
-    |> unmarshal.()
+  defp parse_unary_response({headers, data_list}, unmarshal) do
+    {_, status} = Enum.find(headers, nil, fn(header) -> elem(header, 0) == "grpc-status" end)
+    if status != GRPC.Status.ok do
+      {_, message} = Enum.find(headers, nil, fn(header) -> elem(header, 0) == "grpc-message" end)
+      {:error, %GRPC.RPCError{status: status, message: message}}
+    else
+      result = data_list
+      |> Enum.join("")
+      |> GRPC.Message.from_data
+      |> unmarshal.()
+      {:ok, result}
```

Comment:
I prefer using `result` directly, like what [Erlang rpc](http://erlang.org/doc/man/rpc.html#call-4) does.

Reply:
@tony612 please consider either returning `{:ok, result}` or having both `xxx()` and `xxx!()` forms generated.

It is hard to consume `result` | `{:error, reason}` response with `with ... do` construct.